docs/guides/aws-e2-firewall-hub-and-spoke.md

```diff
@@ -0,0 +1,629 @@
+  })
+}
+
+/* Routing table for spoke private subnet */
+resource "aws_route_table" "spoke_db_private_rt" {
+  vpc_id = aws_vpc.spoke_vpc.id
+  tags = merge(var.tags, {
+    Name = "${local.prefix}-spoke-db-private-rt"
+  })
+}
+
+/* Manage the main routing table for VPC  */
+resource "aws_main_route_table_association" "spoke-set-worker-default-rt-assoc" {
+  vpc_id         = aws_vpc.spoke_vpc.id
+  route_table_id = aws_route_table.spoke_db_private_rt.id
+}
+
+/* Routing table associations for spoke */
+resource "aws_route_table_association" "spoke_db_private_rta" {
+  count          = length(local.spoke_db_private_subnets_cidr)
+  subnet_id      = element(aws_subnet.spoke_db_private_subnet.*.id, count.index)
+  route_table_id = aws_route_table.spoke_db_private_rt.id
+}
+
+/* VPC's Default Security Group */
+resource "aws_security_group" "default_sg" {
+  name        = "${local.prefix}-default-sg"
+  description = "Default security group to allow inbound/outbound from the VPC"
+  vpc_id      = aws_vpc.db_vpc.id
+  depends_on  = [aws_vpc.db_vpc]
+
+  dynamic "ingress" {
+    for_each = local.sg_ingress_protocol
+    content {
+      from_port = 0
+      to_port   = 65535
+      protocol  = ingress.value
+      self      = true
+    }
+  }
+
+  dynamic "egress" {
+    for_each = local.sg_egress_protocol
+    content {
+      from_port = 0
+      to_port   = 65535
+      protocol  = egress.value
+      self      = true
+    }
+  }
+
+  dynamic "egress" {
+    for_each = local.sg_egress_ports
+    content {
+      from_port   = egress.value
+      to_port     = egress.value
+      protocol    = "tcp"
+      cidr_blocks = ["0.0.0.0/0"]
+    }
+  }
+
+  tags = {
+    Environment = var.environment
+  }
+}
+
+resource "databricks_mws_networks" "this" {
+  provider           = databricks.mws
+  account_id         = var.databricks_account_id
+  network_name       = "${local.prefix}-network"
+  security_group_ids = [aws_security_group.default_spoke_sg.id]
+  subnet_ids         = aws_subnet.spoke_db_private_subnet[*].id
+  vpc_id             = aws_vpc.spoke_vpc.id
+}
+
+// Hub VPC
+/* Create VPC */
+resource "aws_vpc" "hub_vpc" {
+  cidr_block           = var.hub_cidr_block
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+  tags = merge(var.tags, {
+    Name = "${local.prefix}-hub-vpc"
+  })
+}
+
+/* Private subnet for Hub TGW Databricks */
+resource "aws_subnet" "hub_tgw_private_subnet" {
+  vpc_id                  = aws_vpc.hub_vpc.id
+  count                   = length(local.hub_tgw_private_subnets_cidr)
+  cidr_block              = element(local.hub_tgw_private_subnets_cidr, count.index)
+  availability_zone       = element(local.availability_zones, count.index)
+  map_public_ip_on_launch = false
+  tags = merge(var.tags, {
+    Name = "${local.prefix}-hub-tgw-private-${element(local.availability_zones, count.index)}"
+  })
+}
+
+/* NAT Public subnet */
```

Comment:
general feeling about this doc - it's a **huge intimidating blob of config**. don't assume that people would be reading blog post to see what things are what.

could you split the guide into a bit more sections with text in between different config blocks, probably just copy-paste those from the blog post :) this way it'll be easier to read. try to make a paragraph of text for every 30-50 lines of config.

Reply:
Probably re-using the source of diagram in the blogpost to create section-specific diagrams would also make this way more friendly and readable.

Reply:
Added some diagrams